src/gennylib/src/PoolManager.cpp

```diff
@@ -62,9 +77,15 @@ mongocxx::pool::entry genny::v1::PoolManager::client(const std::string& name,
 
     Pools& pools = lap.second;
 
-    auto& pool = pools[instance];
+    if (pools.instances.empty() && !pools.encryption) {
+        // set up encryption context when the first instance of this
+        // pool is about to be created.
+        pools.encryption = setupEncryption(name, context, _dryRun);
+    }
```

Comment:
Also not sure how this would handle different pools with different encryption setups.

Reply:
No, this setup will happen on all Pools with encryption enabled. The pools.instances map is empty if the very first instance pool with the given `name` is about to be created.  Since `Default` and `EncryptedPool` are different pool names, there will always be a setupEncryption call when a pool named `EncryptedPool` is first initialized.